### ISSUE TITLE
PDF output does not respect zorder when using imshow

### ISSUE THREAD

**Issue by mrbump** [matplotlib/matplotlib#2517]
I already posted this as a question on [Stack Overflow](http://stackoverflow.com/questions/19312833/matplotlib-pdf-output-does-not-respect-zorder-when-using-imshow).

Minimal example:
```
import numpy as np
import matplotlib.pyplot as plt

img = np.arange(100).reshape((10,10))
plt.imshow(img, extent = [0.25, 0.75, 0.25, 0.75], zorder = 10)
plt.imshow(img, extent = [0.1, 0.9, 0.1, 0.9], zorder = 1)
plt.plot([0, 1], [0, 1], color = 'black', zorder = 5)
plt.axis([0, 1, 0, 1])

plt.savefig('img.png')
```

When exporting to PNG, the output is as expected:
![img](https://f.cloud.github.com/assets/4518755/1324041/aa86df36-34a4-11e3-985e-6c442a4a7403.png)

However, when saving to PDF, the zorder is not respected (the line is plotted over both images):
![img pdf](https://f.cloud.github.com/assets/4518755/1324044/bb529198-34a4-11e3-929c-b1e7001c6654.png)

Is this a bug?

**Comment by tacaswell** [comment]
@Khodeir It looks like you were working on this, did that ever turn into a PR?

**Comment by tacaswell** [comment]
@Stretch97 I could have sworn that we fixed this as part of a different issue...

**Comment by tacaswell** [comment]
This is fixed by #4061

Adding 

```python
matplotlib.rcParams['image.composite_image'] = False
```
to the script suppresses the compositing of images which lets z-order be respected (but makes your pdf files bigger).
